src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const keyToRemove of action.keysToRemove) {
+        delete state[keyToRemove];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const stateKey in state) {
+        delete state[stateKey];
+      }
+    }
```

Comment:
Checklist pt. 2 [CODE STYLE]

You should use `switch` statement when amount of conditions is limited